akka-stream-tests/src/test/scala/akka/stream/scaladsl/FlowInterleaveSpec.scala

```diff
@@ -44,6 +44,15 @@ class FlowInterleaveSpec extends BaseTwoStreamsSetup {
       probe.expectComplete()
     }
 
+    "complete upstream if any of its upstream completes with eagerClose = true" in assertAllStagesStopped {
+      val probe = TestSubscriber.manualProbe[Int]()
+
+      Source(0 to 2).interleave(Source(3 to 5), 2, eagerClose = true).runWith(Sink.fromSubscriber(probe))
+      probe.expectSubscription().request(10)
+      probe.expectNext(0, 1, 3, 4, 2)
+      probe.expectComplete()
+    }
```

Comment:
I didn't mean that the logic is wrong but that it would be good to have test coverage for both the case where segment size is a multiple of the number of elements in the completing source and where it is not. `0 to 2` is `0, 1, 2` so it already isn't on the boundary, but `0 to 1` would be.

And then a left hand `Source` that continues but the passed in `Source` completes early.